src/gui/transferlistsortmodel.cpp

```diff
@@ -86,179 +86,157 @@ void TransferListSortModel::disableTrackerFilter()
 }
 
 bool TransferListSortModel::lessThan(const QModelIndex &left, const QModelIndex &right) const
-{
-    return lessThan_impl(left, right);
-}
-
-bool TransferListSortModel::lessThan_impl(const QModelIndex &left, const QModelIndex &right) const
 {
     Q_ASSERT(left.column() == right.column());
 
-    const auto invokeLessThanForColumn = [this, &left, &right](const int column) -> bool
-    {
-        return lessThan_impl(left.sibling(left.row(), column), right.sibling(right.row(), column));
-    };
-
-    const auto hashLessThan = [this, &left, &right]() -> bool
-    {
-        const TransferListModel *model = qobject_cast<TransferListModel *>(sourceModel());
-        const BitTorrent::InfoHash hashL = model->torrentHandle(left)->hash();
-        const BitTorrent::InfoHash hashR = model->torrentHandle(right)->hash();
-        return hashL < hashR;
-    };
-
     const int sortColumn = left.column();
     const QVariant leftValue = left.data(TransferListModel::UnderlyingDataRole);
     const QVariant rightValue = right.data(TransferListModel::UnderlyingDataRole);
 
     switch (sortColumn)
     {
     case TransferListModel::TR_CATEGORY:
-    case TransferListModel::TR_TAGS:
     case TransferListModel::TR_NAME:
-        if (!leftValue.isValid() || !rightValue.isValid() || (leftValue == rightValue))
-            return invokeLessThanForColumn(TransferListModel::TR_QUEUE_POSITION);
-        return (Utils::String::naturalCompare(leftValue.toString(), rightValue.toString(), Qt::CaseInsensitive) < 0);
+    case TransferListModel::TR_SAVE_PATH:
+    case TransferListModel::TR_TAGS:
+    case TransferListModel::TR_TRACKER:
+        {
+            const auto left = leftValue.toString();
+            const auto right = rightValue.toString();
+
+            if (left != right)
+                return Utils::String::naturalCompare(left, right, Qt::CaseInsensitive) < 0;
+        }
+        break;
+
+    case TransferListModel::TR_AMOUNT_DOWNLOADED:
+    case TransferListModel::TR_AMOUNT_DOWNLOADED_SESSION:
+    case TransferListModel::TR_AMOUNT_LEFT:
+    case TransferListModel::TR_AMOUNT_UPLOADED:
+    case TransferListModel::TR_AMOUNT_UPLOADED_SESSION:
+    case TransferListModel::TR_COMPLETED:
+    case TransferListModel::TR_ETA:
+    case TransferListModel::TR_LAST_ACTIVITY:
+    case TransferListModel::TR_SIZE:
+    case TransferListModel::TR_TIME_ELAPSED:
+    case TransferListModel::TR_TOTAL_SIZE:
+        {
+            const auto left = leftValue.toLongLong();
+            const auto right = rightValue.toLongLong();
+
+            if ((left < 0) && (right < 0))
+                break;
+            else if (right < 0)
```

Comment:
Please remove `else` after "breaking" statements (`return`, `break`, `continue`).

Reply:
Also I see the similar logic is repeated several times. I would extract it to some helper (template) function.

Reply:
>Please remove else after "breaking" statements (return, break, continue).

I'm think the `break` here applies to the if statement and not to the `case`.

>I would extract it to some helper (template) function.

It is impossible to use another function to manipulate the control flow.

Reply:
>I'm think the `break` here applies to the `if` statement and not to the `case`.

🤔
Man, what time is it in your time zone right now? It looks like you've been working hard and it's time for you to rest 😉

Reply:
>It is impossible to use another function to manipulate the intended control flow. At least I don't know how.

Just use it for another job than manipulating control flow:
```c++
if ((left >= 0) || (right >= 0))
    return lessThanFunc(left, right);
break;
```

Reply:
>Just use it for another job than manipulating control flow:

The suggested code isn't correct when `left == right`, in this case it needs to fallback and compare their original positions in list.

>Please remove else after "breaking" statements (return, break, continue).

OK, I'll remove them.

Reply:
>The suggested code isn't correct when `left == right`, in this case it needs to fallback and compare their original positions in list.

Well, then:
```c++
template <typename T>
int compare(T left, T right)
{
    if (((left < 0) && (right < 0)) || (left == right))
         return 0;
    if (right < 0)
        return -1;
    if (left < 0)
        return 1;
    return left - right;
}

// and then:
if (int res = compare(left, right); res != 0)
    return res < 0;
break;
```